tests/substrate_tests/builtins.rs

```diff
@@ -705,4 +705,16 @@ fn my_token() {
 
     runtime.function("test", TokenTest(addr, false).encode());
     assert_eq!(&runtime.vm.output[..], &addr[..]);
+
+    runtime.function(
+        "test",
+        TokenTest(<[u8; 32]>::try_from(&runtime.vm.caller[..]).unwrap(), true).encode(),
+    );
+    assert_eq!(&runtime.vm.caller[..], &runtime.vm.output[..]);
```

Comment:
`runtime.vm.caller` is of type `Account`, which is defined as `type Account = [u8; 32];`. This `try_from..unwrap()` is not really adding anything. In fact I suspect your could do:
` TokenTest(<[u8; 32]>::from(&runtime.vm.caller[..]), true)`

This (and below) is not testing anything different from what we've tested above.

Reply:
This test was a suggestion of [this](https://github.com/hyperledger-labs/solang/issues/679#issuecomment-1060802286) comment, so I thought it would be interesting to add.

Reply:
`<[u8; 32]>::from(&runtime.vm.caller[..])` doesn't work because `the trait `From<&[u8]>` is not implemented for `[u8; 32]``

Reply:
I guess that should be `TokenTest(<[u8; 32]>::from(&runtime.vm.caller), true)` then

Reply:
either way, nothing wrong with adding extra tests :smiley:

Reply:
> I guess that should be `TokenTest(<[u8; 32]>::from(&runtime.vm.caller), true)` then

That doesn't work either because ``` the trait `From<&[u8; 32]>` is not implemented for `[u8; 32]` ```